# Cálculo - Tarefa 02

#### 1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.

In [1]:
import numpy as np

In [2]:
#determinar a função

f = lambda x: x**3

In [7]:
#função para calcular a área da curva - ou seja, a integral 

def calcula_area(a, b, func, num_ret):
    delta = (b-a)/num_ret # largura do retângulo
    x_cols = np.linspace(a, b-delta, num_ret)+delta/2 # Divisão do eixo X
    y_cols = f(x_cols) # alturas dos retângulos
    
    print(f'Número de retângulos = {num_ret}')
    print(f'Delta (dx) = {delta}')
    
    return sum(delta*y_cols)

In [9]:
forma_numerica5 = calcula_area(1, 2, f, 5)
forma_numerica5

Número de retângulos = 5
Delta (dx) = 0.2


3.7350000000000008

In [10]:
forma_numerica20 = calcula_area(1, 2, f, 20)
forma_numerica20

Número de retângulos = 20
Delta (dx) = 0.05


3.7490624999999995

In [11]:
forma_numerica200 = calcula_area(1, 2, f, 200)
forma_numerica200

Número de retângulos = 200
Delta (dx) = 0.005


3.749990625

#### 2- Calcule essa área de forma analítica.

$$\int_{a}^{b} f(x) dx = F(b) - F(a)$$

$$ f(x) = x^3 \rightarrow F(x) = \frac{1}{4} * x^4 + C,$$

$$\int_{1}^{2} x^3 dx = F(2) - F(1) = \left(\frac{1}{4} * 2^4\right) - \left(\frac{1}{4} * 1^4\right) = \frac{16}{4} - \frac{1}{4} = \frac{15}{4} = \textbf{3.75}$$